File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.py
# Last updated: 4/3/2026, 1:59:56 PM
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        curr, best = 0, 0
        for i in nums:
            if i == 1:
                curr += 1
            else:
                curr = 0
            best = max(best, curr)
        return best

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.cpp
// Last updated: 4/3/2026, 2:00:07 PM
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int count = 0, maxCount = 0;
        for (int i : nums) {
            if (i == 1) {
                count++;
                maxCount = max(maxCount, count);
            } else {
                count = 0;
            }
        }
        return maxCount;
    }
};